src/cmt_decode_msgpack.c

```diff
@@ -978,6 +978,9 @@ static int unpack_basic_type_meta(mpack_reader_t *reader, size_t index, void *co
 
     if (CMT_DECODE_MSGPACK_SUCCESS == result) {
         decode_context->map->label_count = cfl_list_size(&decode_context->map->label_keys);
+	if (decode_context->map == NULL || decode_context->map->parent == NULL) {
```

Comment:
if there is a chance that `decode_context->map` is `NULL` ideally we want do it before the previous line otherwise it leads to a corruption `decode_context->map->...`

Reply:
done